File: backend/app/schemas/tag.py
"""
Tag schemas for request/response validation
"""
from typing import Optional
from pydantic import BaseModel, Field


class TagBase(BaseModel):
    """Base tag schema"""
    name: str = Field(..., min_length=1, max_length=50)
    is_system: bool = False


class TagCreate(TagBase):
    """Schema for creating a tag"""
    pass


class TagResponse(TagBase):
    """Schema for tag response"""
    id: str
    use_count: int

    class Config:
        from_attributes = True


class TagListResponse(BaseModel):
    """Schema for tag list response"""
    items: list[TagResponse]
    total: int

File: frontend/src/types/tag.ts
/**
 * Tag type definitions
 */

export interface Tag {
  id: string
  name: string
  is_system: boolean
  use_count: number
}

export interface TagListResponse {
  items: Tag[]
  total: number
}

export interface CreateTagData {
  name: string
  is_system?: boolean
}

export interface TagFilters {
  search?: string
  include_system?: boolean
  popular_only?: boolean
  limit?: number
}
